tests/functional/aws-node-sdk/test/object/corsHeaders.js

```diff
@@ -0,0 +1,756 @@
+        action: s3.uploadPartCopy,
+        params: {
+            Bucket: bucket,
+            CopySource: 'sourceBucket/testSource',
+            Key: objectKey,
+            PartNumber: 1,
+            UploadId: 'testId',
+        },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'POST uploads (create multipart upload)',
+        action: s3.createMultipartUpload,
+        params: { Bucket: bucket, Key: objectKey },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'POST uploadId (complete multipart upload)',
+        action: s3.completeMultipartUpload,
+        params: { Bucket: bucket, Key: objectKey, UploadId: 'testId' },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'POST delete (multi object delete)',
+        action: s3.deleteObjects,
+        params: {
+            Bucket: bucket,
+            Delete: {
+                Objects: [
+                    { Key: objectKey },
+                ],
+            },
+        },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'DELETE bucket',
+        action: s3.deleteBucket,
+        params: { Bucket: bucket },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'DELETE bucket website',
+        action: s3.deleteBucketWebsite,
+        params: { Bucket: bucket },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'DELETE bucket CORS',
+        action: s3.deleteBucketCors,
+        params: { Bucket: bucket },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'DELETE object',
+        action: s3.deleteObject,
+        params: { Bucket: bucket, Key: objectKey },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+    {
+        description: 'DELETE object uploadId (abort multipart upload)',
+        action: s3.abortMultipartUpload,
+        params: { Bucket: bucket, Key: objectKey, UploadId: 'testId' },
+        onlyRun: {
+            checkForCorsHeaders: false,
+            checkNoCorsHeaders: false,
+        },
+    },
+];
+
+// AWS seems to take a bit long so sometimes by the time we send the request
+// the bucket has not yet been created or the bucket has been deleted.
+function _waitForAWS(callback, err) {
+    if (process.env.AWS_ON_AIR) {
+        setTimeout(() => callback(err), 1000);
```

Comment:
problem is AWS responds to request but bucket may still not be finished deleting in background

actually with this much timeout they (some of them) still fail on `AWS_ON_AIR` when I test locally :C I could increase it but I think the person testing can opt to do that if they want to